### PR TITLE
fix: missing computed attributes while clone environment

### PR DESCRIPTION
**Pull request by aiwantaozi**
**Problem:**
Missing computed attributes while cloning environment

**Solution:**
Generate computed attributes for resource while cloning environment

**Related Issue:**
#1909

### CODE REVIEW

**Review by shanewxy — DISMISSED** [review]
LGTM

File: pkg/resource/computed_attributes.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	entity *model.Resource,
+	modelClient *model.Client,
+) (property.Values, error) {
+	// Check.
+	if !entity.TemplateID.Valid() && !entity.ResourceDefinitionMatchingRuleID.Valid() {
+		return nil, fmt.Errorf("failed to generate computed attributes, " +
+			"both template and resource definition machine rule id are empty")
+	}
+
+	if !entity.EnvironmentID.Valid() {
```
Comment (gitlawr) [inline pkg/resource/computed_attributes.go:33]: It panics if any ID is nil

Comment (aiwantaozi) [reply pkg/resource/computed_attributes.go]: updated.

File: pkg/resource/computed_attributes.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+)
+
+// GenComputedAttributes generate computed attributes for resource.
+// Required:
+// entity.EnvironmentID, entity.Attributes
+// entity.TemplateID/entity.ResourceDefinitionMatchingRuleID.
+func GenComputedAttributes(
+	ctx context.Context,
+	entity *model.Resource,
+	modelClient *model.Client,
```
Comment (alexcodelf) [inline pkg/resource/computed_attributes.go:25]: we follow the pattern `func name(context.Context, model.ClientSet`, should we change the order of the params? The modelClient is `*model.Client`, will the function call pass a transaction model client?

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**